backend/cmd/app/main.go

```diff
@@ -0,0 +1,22 @@
+package main
+
+import (
+	"shodo/internal/app"
+
+	_ "shodo/docs"
+
+	"github.com/gin-gonic/gin"
+	swaggerFiles "github.com/swaggo/files"
+	ginSwagger "github.com/swaggo/gin-swagger"
+)
+
+// @title Shodo API
+// @version 1.0
+// @description This is methods declaration for Shodo API
+// @host localhost:8080
+// @BasePath /api/v1
+func main() {
+	r := gin.Default()
```

Comment:
It can be confusing to have gin instantiation and swagger here, and internal/app doing rest of it.
Main func is usually used to setup dependencies and pass them to an app. Then, the app should accept those dependencies and start building from there.

Usually, in main, you would instantiate a database, read the config, do initialization and pass it to the app.
The app assumes you already have everything passed instantiated correctly, and it's responsibility would be to define routes and handlers for your project.
Mixing it up is hard to follow

Reply:
Fixed, moved all initialization into main and rest into app